tests/functional/sources/test_source_configs.py

```diff
@@ -0,0 +1,594 @@
+import pytest
+from dbt.contracts.graph.model_config import SourceConfig
+from dbt.contracts.graph.unparsed import FreshnessThreshold, Time, TimePeriod, Quoting
+from dbt.exceptions import CompilationException
+
+from dbt.tests.util import run_dbt, update_config_file, get_manifest
+
+
+class SourceConfigTests:
+    @pytest.fixture(scope="class", autouse=True)
+    def setUp(self):
+        pytest.expected_config = SourceConfig(
+            enabled=True,
+            # TODO: uncomment all this once it's added to SourceConfig, throws error right now
+            # quoting = Quoting(database=False, schema=False, identifier=False, column=False)
+            # freshness = FreshnessThreshold(
+            #     warn_after=Time(count=12, period=TimePeriod.hour),
+            #     error_after=Time(count=24, period=TimePeriod.hour),
+            #     filter=None
+            #     )
+            # loader = "a_loader"
+            # loaded_at_field = some_column
+            # database = custom_database
+            # schema = custom_schema
+            # # identifier = "seed"  #this doesnt seems to be supported now?
```

Comment:
Good callout. This is necessarily different by table, so it wouldn't make sense to configure for many tables at once.